src/utils/events.utils.ts

```diff
@@ -0,0 +1,24 @@
+import 'reflect-metadata';
+
+import { Event, MicroserviceOptions, TransportType } from '../models';
+import { DECORATORS } from '../models/constants/decorators';
+
+export const buildEventName = (event: Event) => `${event.transport}.${event.event}`;
+
+export const buildEventObject = (options: { event: string; transport?: TransportType; target: object }): Event => {
+  const microserviceOptions: MicroserviceOptions = Reflect.getMetadata(
+    DECORATORS.metadata.microservice.OPTIONS,
+    options.target.constructor,
+  );
+
+  let transportType: TransportType = options.transport || 'internal';
+  if (!options.transport) {
+    if (microserviceOptions) {
+      transportType = microserviceOptions.transport;
+    }
+  }
```

Comment:
Can we use the expression `!options.transport && microserviceOptions` to get rid of one `if`?